pkg/bootflow/types/system_artifact.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+)
+
+type SystemArtifacts map[reflect.Type]SystemArtifact
```

Comment:
Maybe let's use an ID rather then reflect.Type?
This will allow to share code of "firmware part" artifact which all will share same functionality, but just refer to logically different pieces of firmware

Reply:
+ what if some steps will share artifacts? (but say have different types of checks or even similar, that can happen in our abstract world)